apis/v1/gateway_types.go

```diff
@@ -636,10 +665,57 @@ type FrontendTLSValidation struct {
 	// "RefNotPermitted" reason.
 	//
 	// +kubebuilder:validation:MaxItems=8
-	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MinItems=0
 	CACertificateRefs []ObjectReference `json:"caCertificateRefs,omitempty"`
+
+	// FrontendValidationMode defines the mode for validating the client certificate.
+	// There are four possible modes:
+	//
+	// - Request: In this mode, a client certificate is requested
+	//   during the TLS handshake but does not require one.
+	// - RequireAny: In this mode, a client certificate is required during
+	//   the handshake, but the connection is permitted even when the
+	//   client certificate verification fails.
+	// - VerifyIfGiven: In this mode, a client certificate is requested
+	//   but not required. If presented, the certificate must be valid.
+	// - RequireAndVerify: In this mode, a valid client certificate must be
+	//   presented during the handshake and validated
+	//   using CA certificates defined in CACertificateRefs.
+	//
+	// Defaults to RequireAndVerify.
```

Comment:
In previous comments we discussed 3 options for cert validation:
* AllowInvalid
* AllowMissing
* RequireValid
In the current proposal the validation settings are too granular, looking at the fact that they are in Core support. 
IMO we could merge Missing and Invalid into 1 option because in both cases request validation is moved to backend.

Reply:
this eliminates VerifyIfGiven